PWGJE/Tasks/jetSpectraESE.cxx

```diff
@@ -0,0 +1,364 @@
+
+                              {"h_jet_pt_q2_0", "track pT;#it{p}_{T,track} (GeV/#it{c});entries", {HistType::kTH1F, {{200, 0., 200.}}}},
+                              {"h_jet_pt_q2_1", "track pT;#it{p}_{T,track} (GeV/#it{c});entries", {HistType::kTH1F, {{200, 0., 200.}}}},
+                              {"h_jet_pt_in", "track pT;#it{p}_{T,track} (GeV/#it{c});entries", {HistType::kTH1F, {{200, 0., 200.}}}},
+                              {"h_jet_pt_out", "track pT;#it{p}_{T,track} (GeV/#it{c});entries", {HistType::kTH1F, {{200, 0., 200.}}}}
+
+                             }};
+
+  Configurable<float> jetPtMin{"jetPtMin", 5.0, "minimum jet pT cut"};
+  Configurable<float> jetR{"jetR", 0.4, "jet resolution parameter"};
+  Configurable<std::vector<float>> qLimits{"qLimits", {30, 70}, "jet resolution parameter"};
+
+  Configurable<std::string> eventSelections{"eventSelections", "sel8", "choose event selection"};
+  Configurable<std::string> trackSelections{"trackSelections", "globalTracks", "set track selections"};
+
+  int eventSelection = -1;
+  int trackSelection = -1;
+
+  void init(o2::framework::InitContext&)
+  {
+    eventSelection = jetderiveddatautilities::initialiseEventSelection(static_cast<std::string>(eventSelections));
+    trackSelection = jetderiveddatautilities::initialiseTrackSelection(static_cast<std::string>(trackSelections));
+  }
+
+  Filter jetCuts = aod::jet::pt > jetPtMin&& aod::jet::r == nround(jetR.node() * 100.0f);
+
+  void processCollisions(JetCollision const& collision, JetTracks const& tracks)
+  {
+
+    registry.fill(HIST("h_collisions"), 0.5);
+    if (!jetderiveddatautilities::selectCollision(collision, eventSelection)) {
+      return;
+    }
+    registry.fill(HIST("h_collisions"), 1.5);
+    for (auto const& track : tracks) {
+      if (!jetderiveddatautilities::selectTrack(track, trackSelection)) {
+        continue;
+      }
+      registry.fill(HIST("h_track_pt"), track.pt());
+      registry.fill(HIST("h_track_eta"), track.eta());
+      registry.fill(HIST("h_track_phi"), track.phi());
+    }
+  }
+  PROCESS_SWITCH(JetSpectraEseTask, processCollisions, "process self contained collisions", true);
+
+  int spCode(const int qPerc)
+  {
+    if (qPerc < 0)
+      return -1;
+
+    if (qPerc <= qLimits->at(0)) {
+      return 0;
+    } else if (qPerc >= qLimits->at(1)) {
+      return 1;
+    } else {
+      return -1;
+    }
+  }
+  std::string jetplane(float deltaPhi)
+  {
+    std::string s = "";
+    if (deltaPhi < TMath::Pi() / 6.) {
+      s = "in";
+    } else if (deltaPhi > TMath::Pi() / 3.) {
+      s = "out";
+    }
+
+    return s;
+  }
+
+  void processESEDataCharged(JColwESE::iterator const& collision, soa::Filtered<aod::ChargedJets> const& jets, aod::BCsWithTimestamps const&)
+  {
+
+    float vPsi2 = FFitWeights::EventPlane(collision, 2);
+
+    auto qPerc = collision.qPERCFT0C();
```

Comment:
What information from the original collision is missing in JCollisions to make this work?

Reply:
What do you mean exactly? I have joined JCollisions with my own tables in the top:
using JColwESE = soa::Join<aod::JCollisions, aod::CentFT0Cs, aod::qVecFT0Cs, aod::qPercentileFT0Cs>;
this shoudl work, no?

Reply:
Thanks for the quick answer. Yes for now it would be great if you just keep the process functions you need and you can always add more later :)
All the best,
Nima

Reply:
Ah I see, i think i misread the collisions. Ok that is fine then

Reply:
Thanks! I have removed all other process functions for now. :)
Cheers,
Joachim